app/src/main/java/org/kiwix/kiwixmobile/wifi_hotspot/HotspotService.java

```diff
@@ -0,0 +1,188 @@
+package org.kiwix.kiwixmobile.wifi_hotspot;
+
+import android.app.Notification;
+import android.app.NotificationChannel;
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.app.Service;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.os.Binder;
+import android.os.Build;
+import android.os.IBinder;
+import android.util.Log;
+import android.widget.Toast;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.core.app.NotificationCompat;
+import org.kiwix.kiwixmobile.R;
+import org.kiwix.kiwixmobile.utils.Constants;
+import org.kiwix.kiwixmobile.webserver.ServerStateListener;
+import org.kiwix.kiwixmobile.webserver.WebServerHelper;
+import org.kiwix.kiwixmobile.webserver.ZimHostActivity;
+
+import static org.kiwix.kiwixmobile.webserver.ZimHostActivity.ACTION_IS_HOTSPOT_ENABLED;
+import static org.kiwix.kiwixmobile.webserver.ZimHostActivity.ACTION_START_SERVER;
+import static org.kiwix.kiwixmobile.webserver.ZimHostActivity.ACTION_STOP_SERVER;
+import static org.kiwix.kiwixmobile.webserver.ZimHostActivity.ACTION_TURN_OFF_AFTER_O;
+import static org.kiwix.kiwixmobile.webserver.ZimHostActivity.ACTION_TURN_ON_AFTER_O;
+import static org.kiwix.kiwixmobile.webserver.ZimHostActivity.SELECTED_ZIM_PATHS_KEY;
+
+/**
+ * HotspotService is used to add a foreground service for the wifi hotspot.
+ * Created by Adeel Zafar on 07/01/2019.
+ */
+
+public class HotspotService extends Service {
+  private static final int HOTSPOT_NOTIFICATION_ID = 666;
+  private static final String ACTION_STOP = "hotspot_stop";
+  private static final String TAG = "HotspotService";
+  private WifiHotspotManager hotspotManager;
+  private BroadcastReceiver stopReceiver;
+  private NotificationManager notificationManager;
+  private NotificationCompat.Builder builder;
+  private ServerStateListener serverStateListener;
+  private final IBinder serviceBinder = new HotspotBinder();
+  private WebServerHelper webServerHelper;
+
+  @Override public void onCreate() {
+
+    super.onCreate();
+
+    hotspotManager = new WifiHotspotManager(this);
```

Comment:
injection here is a bit more complex as you will want to define a service component. I have a particular hate for the `dagger-android` module and have avoided it but you can try following [this](https://github.com/googlesamples/android-architecture-components/issues/253)? It could get quite complex very quick.

Reply:
The other option is to avoid `dagger-android` and just create your own ServiceComponent and call
`KiwixApplication.getComponent().serviceComponent.service(this).build().inject(this)`